### ISSUE TITLE
JSON worker support

### ISSUE THREAD

**Issue by johnynek** [bazelbuild/rules_scala#1361]
these rules were early (the earliest?) adopters of the worker protocol, which uses protobuf, but the proto dependency has caused a lot of issues:

#779 #1254 #1305

but recently, bazel has supported a json protocol:

https://blog.bazel.build/2020/11/11/json-workers.html

I am skeptical that such a simple json parse and generation could possibly harm performance (compared to the runtime of the compiler itself) so given the history of problems, this migration seems worth attempting.

I can probably find time to send a PR if this seems like something the current maintainers would accept.

**Comment by johnynek** [comment]
this could also sidestep the need to solve: #1330

**Comment by liucijus** [comment]
Yes, I like the idea to simplify workers using json. It would be great if we could keep protobuf implementation for some time too and than remove it if json implementation performs well.

PRs are more than welcome!
